Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -329,38 +390,61 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
         }
     }
 
+    bool matchedElementIsGuardContainer = isGuardContainer(*matchedElement);
+
+    if (isOriginalMatch && matchedElementIsGuardContainer) {
+        return { {
+            InteractionRegion::Type::Guard,
+            elementIdentifier,
+            bounds
+        } };
+    }
+
     // The parent will get its own InteractionRegion.
-    if (!isOriginalMatch && !isPhoto && !isInlineNonBlock && !renderer.style().isDisplayTableOrTablePart())
+    if (!isOriginalMatch && !matchedElementIsGuardContainer && !isPhoto && !isInlineNonBlock && !renderer.style().isDisplayTableOrTablePart())
         return std::nullopt;
 
+    RefPtr<Image> iconImage;
+    std::optional<std::pair<SVGSVGElement*, SVGGraphicsElement*>> svgClipElements;
+    if (!bigEnoughForPhoto)
+        iconImage = findIconImage(regionRenderer);
+    if (!iconImage)
+        svgClipElements = findSVGClipElements(regionRenderer);
+
     auto rect = bounds;
     float cornerRadius = 0;
     OptionSet<InteractionRegion::CornerMask> maskedCorners { };
     std::optional<Path> clipPath = std::nullopt;
 
-    if (const auto& renderShape = dynamicDowncast<LegacyRenderSVGShape>(regionRenderer)) {
-        auto& svgElement = renderShape->graphicsElement();
-        auto path = svgElement.toClipPath();
+    if (iconImage && originalElement) {
+        auto imageRect = LayoutRect(rect);
+        auto shape = Shape::createRasterShape(iconImage.get(), 0, imageRect, imageRect, WritingMode::HorizontalTb, 0);
```

Comment:
I guess you don't actually care about the writing mode?

Reply:
nope, comment worthy?

Reply:
Please don't hide smart pointers with `auto`. Should use `Ref` here.